doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2006 @@
+        use stdlib_bitsets
+        type(bitset_large) :: set0, set1
+        call set0 % init(166)
+        call set1 % init(166)
+        call and( set0, set1 ) ! none none
+        if ( none(set0) ) write(*,*) 'First test of AND worked.'
+        call set0 % not()
+        call and( set0, set1 ) ! all none
+        if ( none(set0) ) write(*,*) 'Second test of AND worked.'
+        call set1 % not()
+        call and( set0, set1 ) ! none all
+        if ( none(set0) ) write(*,*) 'Third test of AND worked.'
+        call set0 % not()
+        call and( set0, set1 ) ! all all
+        if ( all(set0) ) write(*,*) 'Fourth test of AND worked.'
+    end program demo_and
+```
+
+### `and_not` - Bitwise `and` of one bitset with the negation of another
+
+#### Status
+
+Experimental
+
+#### Description
+
+Sets the bits of `set1` to bitwise `and` of the bits of `set1` with
+the bitwise negation of the corresponding bits of `set2`. Note that
+`set1` and `set2` must have the same number of bits, otherwise the
+result is undefined.
+
+#### Syntax
+
+`call [[stdlib_bitsets(module):and_not(interface)]](set1, set2)`
+
+#### Class
+
+Elemental subroutine.
+
+#### Arguments
+
+`set1`: shall be a scalar `bitset_64` or `bitset_large` variable. It
+is an `intent(inout)` argument. On return the values of the bits in
+`set1` are the bitwise `and` of the original bits in `set1` with the
+corresponding negation of the bits in `set2`.
+
+`set2`: shall be a scalar expression of the same type as `set1`. It is
+an `intent(in)` argument. Note that it should also have the same
+number of bits as `set1` otherwise the result is undefined.
+
+#### Example
+
+```fortran
+    program demo_and_not
+        use stdlib_bitsets
+        type(bitset_large) :: set0, set1
+        call set0 % init(166)
+        call set1 % init(166)
+        call and_not( set0, set1 ) ! none none
+        if ( none(set0) ) write(*,*) 'First test of AND_NOT worked.'
+        call set0 % not()
+        call and_not( set0, set1 ) ! all none
+        if ( all(set0) ) write(*,*) 'Second test of AND_NOT worked.'
+        call set0 % not()
+        call set1 % not()
+        call and_not( set0, set1 ) ! none all
+        if ( none(set0) ) write(*,*) 'Third test of AND_NOT worked.'
+        call set0 % not()
+        call and_not( set0, set1 ) ! all all
+        if ( none(set0) ) write(*,*) 'Fourth test of AND_NOT worked.'
+    end program demo_and_not
+```
+
+### `any` - determine whether any bits are set
+
+#### Status
+
+Experimental
+
+#### Description
+
+Determines whether any bits are set in `self`.
+
+#### Syntax
+
+`result = self % [[bitset_type(class):any(bound)]]()`
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+`self`: shall be a scalar expression of class `bitset_type`. It is an
+`intent(in)` argument.
+
+#### Result character
+
+The result is a default logical scalar.
+
+#### Result value
+
+The result is `.true.` if any bits in `self` are set, otherwise it
+is `.false.`.
```

Comment:
These 2 sections could be merged. Currently it is confusing.
```suggestion
#### Result value

The result is a default logical scalar. The result is `.true.` if any bits in `self` are set, otherwise it
is `.false.`.
```

Reply:
OK